src/test/java/GmailActionableEmailsServletTest.java

```diff
@@ -0,0 +1,179 @@
+
+import com.google.api.services.gmail.model.Message;
+import com.google.api.services.gmail.model.MessagePart;
+import com.google.api.services.gmail.model.MessagePartHeader;
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.sps.model.GmailClient;
+import com.google.sps.model.GmailClientFactory;
+import com.google.sps.servlets.GmailActionableEmailsServlet;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+/** Tests the GmailActionableEmailsServlet */
+@RunWith(JUnit4.class)
+public class GmailActionableEmailsServletTest extends AuthenticatedServletTestBase {
+  private GmailClient gmailClient;
+  private GmailActionableEmailsServlet servlet;
+
+  private static final Gson gson = new Gson();
+
+  // spaces should be parsed correctly by request.getParameter if query string is properly encoded
+  private static final String SUBJECT_LINE_WORDS_STRING = "Action Word One, ActionWordTwo";
+
+  private static final GmailClient.MessageFormat messageFormat = GmailClient.MessageFormat.FULL;
+
+  int DEFAULT_N_DAYS = 7;
+  int NEGATIVE_N_DAYS = -1;
+
+  List<Message> SOME_MESSAGES =
+      ImmutableList.of(new Message().setId("messageFour"), new Message().setId("messageFive"));
+
+  /**
+   * Auxiliary method to get a Message payload (with a "From" header) given a sender's email. From
+   * header in the form of: "From": "SenderName <email@email.com>"
+   *
+   * @param email the sender's email
+   * @param contactName the name of the sender
+   * @return a MessagePart instance that can be used as the payload of a Message
+   */
+  static MessagePart generateMessagePayload(String email, String contactName) {
+    return new MessagePart()
+        .setHeaders(
+            Collections.singletonList(
+                new MessagePartHeader()
+                    .setName("From")
+                    .setValue(String.format("%s <%s>", contactName, email))));
+  }
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+    GmailClientFactory gmailClientFactory = Mockito.mock(GmailClientFactory.class);
+    gmailClient = Mockito.mock(GmailClient.class);
+    servlet = new GmailActionableEmailsServlet(authenticationVerifier, gmailClientFactory);
+    Mockito.when(gmailClientFactory.getGmailClient(Mockito.any())).thenReturn(gmailClient);
+  }
+
+  @Test
+  public void subjectLineWordsNull() throws Exception {
+    Mockito.when(request.getParameter("unreadOnly")).thenReturn(String.valueOf(true));
+    Mockito.when(request.getParameter("nDays")).thenReturn(String.valueOf(DEFAULT_N_DAYS));
+
+    servlet.doGet(request, response);
+    Assert.assertEquals(400, response.getStatus());
```

Comment:
If I understand correctly, 400 is to show that an error has occurred. I think it would be nice to have a comment explaining that since it's not immediately clear why an error is occurring.

Reply:
400 is to show an illegal argument exception - the name of the method specifies that the subjectWordPhrases is null (which should imply the error here)

Reply:
Fair enough - thanks for clarifying!

Reply:
`RESOLVED`